packages/connect/e2e/tests/device/cancel.test.ts

```diff
@@ -0,0 +1,134 @@
+import { SetupEmu } from '@trezor/trezor-user-env-link';
+import TrezorConnect from '../../../src';
+
+const { getController, setup, initTrezorConnect } = global.Trezor;
+
+const getAddress = async (showOnTrezor: boolean) => {
+    return TrezorConnect.getAddress({
+        path: "m/44'/1'/0'/0/0",
+        showOnTrezor,
+    });
+};
+
+const passphraseHandler = (value: string) => () => {
+    TrezorConnect.uiResponse({
+        type: 'ui-receive_passphrase',
+        payload: {
+            passphraseOnDevice: false,
+            value,
+        },
+    });
+    TrezorConnect.removeAllListeners('ui-request_passphrase');
+};
+
+const assertGetAddressWorks = async () => {
+    // validate that further communication is possible without any glitch
+    TrezorConnect.on('ui-request_passphrase', passphraseHandler('a'));
+    const getAddressResponse = await getAddress(false);
+    expect(getAddressResponse.success).toEqual(true);
+};
+
+describe('TrezorConnect.cancel', () => {
+    const controller = getController();
+
+    const setupTest = async ({ setupParams }: { setupParams: SetupEmu }) => {
+        await setup(controller, setupParams);
+        await TrezorConnect.dispose();
+        await initTrezorConnect(controller, { debug: false });
+    };
+
+    afterAll(async () => {
+        controller.dispose();
+        await TrezorConnect.dispose();
+    });
+
+    // the goal is to run this test couple of times to uncover possible race conditions/flakiness
+    [0, 1, 10, 100, 300].forEach(delay => {
```

Comment:
it looks like that higher delays are usually ok but the lower ones tend to be flaky